PluralKit.Bot/Commands/Message.cs

```diff
@@ -22,23 +24,70 @@ public class ProxiedMessage
     private static readonly Duration EditTimeout = Duration.FromMinutes(10);
 
     // private readonly IDiscordCache _cache;
+    private readonly ModelRepository _repo;
     private readonly IClock _clock;
+    private readonly IMetrics _metrics;
 
     private readonly EmbedService _embeds;
     private readonly LogChannelService _logChannel;
     private readonly DiscordApiClient _rest;
     private readonly WebhookExecutorService _webhookExecutor;
+    private readonly ProxyService _proxy;
 
     public ProxiedMessage(EmbedService embeds, IClock clock,
-                          DiscordApiClient rest,
+                          DiscordApiClient rest, IMetrics metrics, ModelRepository repo, ProxyService proxy,
                           WebhookExecutorService webhookExecutor, LogChannelService logChannel, IDiscordCache cache)
     {
         _embeds = embeds;
         _clock = clock;
         _rest = rest;
         _webhookExecutor = webhookExecutor;
+        _repo = repo;
         _logChannel = logChannel;
         // _cache = cache;
+        _metrics = metrics;
+        _proxy =  proxy;
+    }
+
+    public async Task ReproxyMessage(Context ctx)
+    {
+        var msg = await GetMessageToEdit(ctx);
+
+        if (ctx.System.Id != msg.System?.Id)
+            throw new PKError("Can't reproxy a message sent by a different system.");
+
+        // Bail if the message is more than 2 minutes old
+        var msgTimestamp = Instant.FromUnixTimeMilliseconds((long)(msg.Message.Mid >> 22) + 1420070400000);
+        if (msgTimestamp.Plus(Duration.FromSeconds(60 * 2)) < SystemClock.Instance.GetCurrentInstant())
+            throw new PKError("The message is too old to be reproxied.");
+
+        // Get target member ID
+        var target = await ctx.MatchMember();
+        var previousPtr = ctx.Parameters._ptr;
+        if (target == null)
+            throw new PKError("Could not find a member to reproxy the message with.");
+
+        // Fetch members and get the ProxyMember for `target`
+        List <ProxyMember> members;
+        using (_metrics.Measure.Timer.Time(BotMetrics.ProxyMembersQueryTime))
+            members = (await _repo.GetProxyMembers(ctx.Author.Id, msg.Message.Guild!.Value)).ToList();
+        var match = members.Find(x => x.Id == target.Id);
+        if (match == null)
+            throw new PKError("Could not find a member to reproxy the message with.");
+
+        try
+        {
+            await _proxy.ExecuteReproxy(ctx.Message, msg.Message, match);
+
+            if (ctx.Guild == null)
+                await _rest.CreateReaction(ctx.Channel.Id, ctx.Message.Id, new Emoji { Name = Emojis.Success });
+            if ((await ctx.BotPermissions).HasFlag(PermissionSet.ManageMessages))
+                await _rest.DeleteMessage(ctx.Channel.Id, ctx.Message.Id);
```

Comment:
this is already done by HandleProxyExecutedActions, you just need to delete the original proxied message

Reply:
this is done here so that this commands acts the same way as `pk;edit` - deleting the trigger if in a channel, marking it as successful with an emoji in DMs. figured it's best to have consistency between the message-editing-related commands?

(that is also why we added the "don't delete the trigger" override flag in HandleProxyExecutedActions)

Reply:
ohh, okay, that makes sense